### PR TITLE
src: run clippy and fix issues

### PR DESCRIPTION
**Pull request by jaudiger**
Very similar to https://github.com/nushell/nu-ansi-term/pull/58 but on src folder this time

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -34,20 +34,22 @@ pub fn sub_string(start: usize, len: usize, strs: &AnsiStrings) -> Vec<AnsiStrin
...
-        s += i.string.deref();
+    for i in strs.0 {
+        s += &*i.string;
     }
 
     s
 }
 
-/// Return the unstyled length of AnsiStrings. This is equaivalent to `unstyle(strs).len()`.
+/// Return the unstyled length of `AnsiStrings`. This is equivalent to `unstyle(strs).len()`.
```
Comment (jaudiger) [inline src/util.rs:48]: A typo was hidden here: `equaivalent `

**Comment by sholderbach** [comment]
Before we consider this, we are currently lacking a MSRV checking CI step. 
While Nushell itself is pretty bleeding edge we have a bunch of large crates that are more conservative about MSRV as [dependents](https://crates.io/crates/nu_ansi_term/reverse_dependencies) (e.g. [`tracing-subscriber` is on 1.63.0](https://github.com/tokio-rs/tracing/blob/690a9a68098aa1eb86d1c5c4aad085ef12245bca/tracing-subscriber/Cargo.toml#L23)). 
Out of courtesy it may be best to make sure we can maintain the MSRV until there are clear needs for more recent language features (a contentious one could be `const`ification)

If this is not MSRV bumping I would be fine with landing it, but we will probably wait on shipping until a necessary bugfix/new feature arrives.

**Comment by jaudiger** [comment]
I checked this morning of the two PRs I opened were raising the MSRV with the tool `cargo-msrv`. From my testing, the MSRV is not affected on both branches:

```bash
cargo msrv
Fetching index
Determining the Minimum Supported Rust Version (MSRV) for toolchain aarch64-apple-darwin
Using check command cargo check
Check for toolchain '1.67.1-aarch64-apple-darwin' succeeded

Check for toolchain '1.61.0-aarch64-apple-darwin' failed with:
┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ error: package `nu-ansi-term v0.50.0 (/Users/jaudiger/Development/git-repositories/jaudiger/nu-ansi-term)` cannot be built because it requires rustc 1.62.1 or newer, while the currently active rustc version is 1.61.0                                                                                                                          │
└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
Check for toolchain '1.64.0-aarch64-apple-darwin' succeeded
Check for toolchain '1.62.1-aarch64-apple-darwin' succeeded
   Finished The MSRV is: 1.62.1
```

Please let me know if there is anything else I could do. If you prefer to split the PR and focus on one work at a time, I'm also okay.

**Comment by kubouch** [comment]
When I run `cargo clippy` in the `examples/` folder, nothing shows up.

**Pull request closed without merging**